src/api/voting.ts

```diff
@@ -184,15 +193,22 @@ export namespace VotingApi {
      * Fetch the full state of the given processId on the Vochain
      * @param processId
      * @param gateway
+     * @param params
      */
-    export function getProcessState(processId: string, gateway: IGatewayDVoteClient): Promise<ProcessState> {
+    export function getProcessState(processId: string, gateway: IGatewayClient, params: { skipArchive?: boolean } = {}): Promise<ProcessState> {
         if (!processId) return Promise.reject(new Error("Empty process ID"))
         else if (!gateway) return Promise.reject(new Error("Invalid gateway client"))
 
         return gateway.sendRequest({ method: "getProcessInfo", processId })
+            .catch((error) => {
+                const message = error.message ? "Could not retrieve the process info: " + error.message : "Could not retrieve the process info"
+                if (params.skipArchive || !message.includes("No data found for this key")) {
+                    throw new Error(message)
```

Comment:
```suggestion
                    throw error
```

This way we preserve the original stack trace, no need to create a new Error

Reply:
In general, I would only build the `message` variable when we do actually need to throw an error. Otherwise, this is work that we may not need to do.

Reply:
In addition, I would also create a kind of `NoSuchProcessError` error, so that we don't need to explicitly compare with `"No data found for this key"`

Instead, do a check like `error instanceof NoSuchProcessError`